### ISSUE TITLE
Dashboard API console 'run as' doesn't appear to do anything, especially when ACL is concerned

### ISSUE THREAD

**Issue by MobileVet** [parse-community/parse-dashboard#449]
We utilize ACL to lock down access to our data.  This is creating a bit of a challenge in the testing phase as the API console doesn't seem to allow setting the user.  Even if I include a User objectId inside the 'Run as...', request.user.id is undefined in the Cloud Code.

Is it expected that 'request.user.id' is undefined in this situation?

**Comment by Abderezai** [comment]
:+1:

**Comment by MobileVet** [comment]
Little surprised no one else has dealt with this... any thoughts on how we would go about adding this functionality?  If this isn't what 'Run as...' is intended for... what is it supposed to do?

**Comment by aziz92** [comment]
@MobileVet a bit late but having the same issue. How you find a way around this? Quite surprised that this bug hasn't been fixed for almost a year. Losing hope in Parse, seems like no one cares.

**Comment by flovilmart** [comment]
@aziz92 for sure, if everybody was as active as you, everything would be way better. Not sure if you see the volume of activity there’s been in the past year, but yeah, we’ve been busy. 
Sometimes issues don’t get fixed for a various lot of reasons. 

What’s surprising, it’s that within a year, you didn’t fixed that bug that’s been bothering you. 

Our responsibility as maintainers doesn’t extend to fixing all open issues. Our responsibility is to be as welcoming, open and helpful as we can. 

We’ll gladly accept your PR as always.

**Comment by Abderezai** [comment]
@flovilmart 
Appreciate your hard work. I personally wouldn't have the confidence to miss with the code base that so many ppl depend on. But I am more than willing to financially contribute to keeping parse progressing forward. It takes time to maintain this, and I am making an income using parse. I can help set something up where ppl contribute to keep this going. Or is this blasphemous?

**Comment by flovilmart** [comment]
@Abderezai this is not about money, finance, we all have day jobs, use parse in our own projects. Paying me to fix issue would actually not get you anywhere. But feel free to pay a consultant to open a Pr if you think that's the way forward. 

I can only encourage you to LOOK at what the issue is, we're here, to review pull requests so it's unlikely you'll break anything.

**Comment by aziz92** [comment]
@flovilmart I just found about about the bug today, so it hasn't been a year. Not sure why that is surprising to you. The original guy posted this a year ago and that's what I was referring to.  

Anyways, I will be looking into this bug now but I'm new to Parse and i need to study the code base before I can even think about touching it. 

With that being said, I appreciate the work you (and lot of others) have put into this. My post came off as offensive, I can see that and I apologize for it.

**Comment by MobileVet** [comment]
Interestingly, it has been working fine for us on 1.0.22.  Not sure what version that changed in honestly, but one time I tried it and it worked.  I use it regularly now otherwise I would have looked into resolving it.

I realize 1.0.22 is as old as dirt now...  We have been heads down building out the project and haven't wanted to change anything while we did.  Now that we are stable and solid we will be bringing libraries up to the latest.  Will post back if I find the recent version isn't working.
